apps/arena/lib/arena/game_updater.ex

```diff
@@ -557,12 +505,208 @@ defmodule Arena.GameUpdater do
     end)
   end
 
+  ##########################
+  # End Game Initialization
+  ##########################
+
+  ##########################
+  # Game flow. Actions executed in every tick.
+  ##########################
+
+  defp move_players(
+         %{
+           players: players,
+           ticks_to_move: ticks_to_move,
+           external_wall: external_wall,
+           obstacles: obstacles,
+           power_ups: power_ups
+         } = game_state
+       ) do
+    moved_players =
+      players
+      |> Physics.move_entities(
+        ticks_to_move,
+        external_wall,
+        obstacles
+      )
+      |> update_collisions(players, power_ups)
+
+    %{game_state | players: moved_players}
+  end
+
+  # Remove exploded projectiles
+  # Update expired to explode status
+  defp update_projectiles_status(%{projectiles: projectiles} = game_state) do
+    updated_projectiles =
+      projectiles
+      |> Map.reject(fn {_projectile_id, projectile} ->
+        projectile.aditional_info.status == :EXPLODED
+      end)
+      |> Enum.reduce(%{}, fn {projectile_id, projectile}, acc ->
```

Comment:
You are iterating twice over `projectiles`, should we do this both things in `Enum.reduce/3`?

Sort of like
```elixir
Enum.reduce(projectiles, projectiles, fn {projectile_id, projectile}, acc ->
  case projectile.aditional_info.status do
    :EXPLODED -> Map.delete(acc, projectile_id)
    :EXPIRED -> put_in(acc, [projectile_id, :aditional_info, :status], :EXPLODED))
    _ -> acc
  end
end)

Reply:
Yeah, looks much better. Done!